### ISSUE TITLE
PartySocket v0.0.10 throws react useRef error

### ISSUE THREAD

**Issue by sammdec** [partykit/partykit#486]
I just upgrade to use partysocket v0.0.10 and found that whern instantiating a basic instance of the `usePartySocket` hook, it throws a cannot read properties of null error both on the server render and the client, my assumption is that this has something to do with the underlying PartyKit ws server not ebing defined (maybe a race condition?). It works fine in `0.0.9`, I tried looking at the diff but couldnt find anything obiovus that made any changes, (hence why i thought it might be a subtle race condition somewhere).

Here is a simple repro app https://codesandbox.io/p/sandbox/practical-chatterjee-xps77q?file=%2Fapp%2Fpage.tsx%3A13%2C7

**Comment by threepointone** [comment]
damn I think this might be the reason https://github.com/partykit/partykit/blob/4befdfa576588de2cba920ad3ce1bbdb7fd4e3e6/packages/partysocket/package.json#L42-L44 I suspect it's installed multiple versions of react. Maybe we should change that version to "*".

**Comment by threepointone** [comment]
This might fix it https://github.com/partykit/partykit/pull/487

**Comment by sammdec** [comment]
I am using the exact same version of React though if that makes any difference

**Comment by sammdec** [comment]
> This might fix it #487

Worth a shot 👍

**Comment by threepointone** [comment]
hmm we're having trouble deploying betas right now, but please stay pinned to 0.0.9 until we roll it out

**Comment by sammdec** [comment]
All good, no rush!

**Comment by threepointone** [comment]
Try now with `npm install partysocket@beta`? 

(no rush, whenever you get the chance)

**Comment by sammdec** [comment]
Updated the anove codesandbox as well and can repro that the same issue is occurring

**Comment by threepointone** [comment]
sorry, was afk, looking into it now

**Comment by threepointone** [comment]
Looks like you caught an actual bug with our bundler. tsup only excludes things mentioned under dependencies/devDependencies. So when we moved it to optionalDependencies, it started including it, resulting in 2 reacts in the bundle. Further, because of the way package-lock.json works, I think was still picking up an older version. So I just landed a new beta with a proper fix, forked your sandbox and tried it with the explicit new beta version, and it appears to be working now https://codesandbox.io/p/sandbox/recursing-tree-skktd4?file=%2Fpackage.json%3A24%2C34

So I recommend you stay with 0.0.9, and then pick up the new release 0.0.11 when it's out (which should be before end of week, probably sooner)

Thanks so much for reporting this, and sorry for the trouble!
